src/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ChangeEvent } from 'react';
+import { InlineField, Input, SecretInput } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+// import { MyDataSourceOptions, MySecureJsonData } from '../types';
+import { MyDataSourceOptions } from '../types';
+
+interface Props extends DataSourcePluginOptionsEditorProps<MyDataSourceOptions> { }
+
+export function ConfigEditor(props: Props) {
+  const { onOptionsChange, options } = props;
+
+  const onAwsAccessKeyIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const jsonData = {
+      ...options.jsonData,
+      awsAccessKeyId: event.target.value,
+    };
+    onOptionsChange({ ...options, jsonData });
+  };
+
+  const onAwsSecretAccessKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const jsonData = {
```

Comment:
This looks like there is some boiler plate here, but this seems funky. If this is a state, I'm curious how it's being managed and set. Doing the spread on options is going to be weird, you can get a race condition on the frontend.